### PR TITLE
[Enhancement] Prevent deadlock in IterBasedRunner

### PR DESCRIPTION
**Pull request by RangiLyu**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Prevent potential deadlock in IterBasedRunner.

## Modification

Add `time.sleep(2)` in `IterLoader` when epoch transition.

## BC-breaking (Optional)

None

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Review by zhouzaida — APPROVED** [review]

File: mmcv/runner/iter_based_runner.py [hunk 1]
```diff
@@ -34,6 +34,7 @@ def __next__(self):
             self._epoch += 1
             if hasattr(self._dataloader.sampler, 'set_epoch'):
                 self._dataloader.sampler.set_epoch(self._epoch)
+            time.sleep(2)  # Prevent possible deadlock during epoch transition
```
Comment (MeowZheng) [inline mmcv/runner/iter_based_runner.py:37]: In my experience, I never met this problem, and 2 seconds might be unfriendly to small datasets.

Comment (RangiLyu) [reply mmcv/runner/iter_based_runner.py:37]: We met this problem in MME.

Comment (MeowZheng) [reply mmcv/runner/iter_based_runner.py:37]: OK. It doesn't have much effect on me. Might hear others' opinions.

**Review by MeowZheng — APPROVED** [review]

File: mmcv/runner/iter_based_runner.py (hunk 1 above)
Comment (nbei) [reply mmcv/runner/iter_based_runner.py:37]: In fact, I have NOT met any deadlock caused by the dataloader. If this modification can solve your problem in training, we can merge it.

**Review by nbei — APPROVED** [review]

**Comment by Junjun2016** [comment]
Which codebase and dataset?

**Review by ZwwWayne — APPROVED** [review]

File: mmcv/runner/iter_based_runner.py (hunk 1 above)
Comment (JihwanEom) [reply mmcv/runner/iter_based_runner.py:37]: Hi @RangiLyu, could you explain what is MME? Can I reproduce deadlock?

**Pull request merged**